### PR TITLE
Require purposes

### PR DESCRIPTION
**Pull request by adamnfish**
This PR includes changes in this previous PR, let's review and merge that first:
https://github.com/guardian/consent-logs/pull/13

The Guardian/PECR purposes (functional, performance and personalisedAdvertising) are now required in the record body. Previously we were ensuring the records were valid keys, but not that all keys were present.

This also does a little renaming, I've tried to make it a little clearer where we're talking about IAB bits vs consent in general.

### CODE REVIEW

File: src/amp.spec.ts [hunk 1]
```diff
@@ -240,14 +240,84 @@ describe('fullAmpConsent', () => {
...
+
+    test('creates a fully consented IAB string', () => {
+      const fullConsent = consentModelFrom('abc', true);
+      expect(parseJson(JSON.stringify(fullConsent))).toEqual(fullConsent);
+    });
+
+    test('creates a fully consented IAB string', () => {
+      const fullConsent = consentModelFrom('abc', true);
+      expect(parseJson(JSON.stringify(fullConsent))).toEqual(fullConsent);
+    });
```
Comment (janua) [inline src/amp.spec.ts]: This test is in here twice?

File: src/model.spec.ts [hunk 2]
```diff
@@ -320,6 +330,16 @@ describe('isValidPurposes', () => {
         invalidKeysPurposesListArbitrary,
         (invalidPurposes) => !isValidPurposes(invalidPurposes)));
   });
+
+  test('does not accept an object missing required purposes', () => {
+    const removeProperty =
+        (propKey: any, {[propKey]: propValue, ...rest}: any) => rest;
```
Comment (janua) [inline src/model.spec.ts]: This is nice

**Comment by janua** [comment]
:+1:

File: src/model.spec.ts (hunk 2 above)
Comment (adamnfish) [reply src/model.spec.ts]: Only works with the no strict annotation though, alas. I couldn't work out how to do it with proper types :-(

**Pull request merged**